addons/sale_loyalty/models/sale_order.py

```diff
@@ -1107,7 +1107,7 @@ def __try_apply_program(self, program, coupon, status):
                 all_points = [p for p in all_points if p]
                 partner = False
                 # Loyalty programs and ewallets are nominative
-                if program.is_nominative:
+                if program.is_nominative or program.program_type == 'next_order_coupons':
```

Comment:
what happens if the program meets this condition but the partner is False?

Reply:
The `partner` variable is initially set to `False` before the condition is checked. If the condition is met, the partner will be set to `self.partner_id.id` and if not then it will remain `False`

Reply:
what I mean is self.partner_id is always set ?

Reply:
Yes, `self.partner_id` is always set since it is a required field in the `sale.order` model.